src/v/lsm/block/builder.cc

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "lsm/block/builder.h"
+
+#include <limits>
+#include <utility>
+
+namespace lsm::block {
+
+builder::builder()
+  : builder(builder::options{}) {}
+
+builder::builder(builder::options opts) {
+    _restarts.reserve(opts.restart_interval);
```

Comment:
I'm confused about why we're reserving the number of restarts based on the interval. I would've expected the reservation to be some `<expected num restarts> ~= <expected block size> / <restart interval>`. Or at the very least, I would have expected the number of restarts to be inversely correlated with the restart interval (higher interval, fewer restarts). Am I missing something?

Reply:
Yes good point. I removed this as it was incomplete